cmd/notation/push.go

```diff
@@ -70,7 +71,12 @@ func runPush(command *cobra.Command, opts *pushOpts) error {
 			return err
 		}
 		// pass in nonempty annotations if needed
-		sigDesc, _, err := sigRepo.PutSignatureManifest(command.Context(), sig, manifestDesc, make(map[string]string))
+		// TODO: understand media type in a better way
```

Comment:
Both push and verify will need to speculate format now not only for caching. 
For verifying, the new verification workflow with trust store and trust policy doesn't need to speculate format because it will pull from remote.
https://github.com/notaryproject/notation-go/blob/3d8de5f05facb36c97df60de6e49c936e5ac6b41/verification/verifier.go#L100-L116

For pushing, there is a flag named `signatures`, which is the paths to some signatures. CLI will first read the file and push its content to the remote. 
https://github.com/notaryproject/notation/blob/main/specs/notation-cli.md#push

https://github.com/notaryproject/notation/blob/4666267499a11f5c56af7211fee36a27e78cd582/cmd/notation/push.go#L52-L71

So, for RC1, verify doesn't need speculate, but we still need to speculate format if user pushes a signature from local file system unless we abandon this feature.
If still necessary, I can open an issue to track the push problem.

Reply:
Yeah unless there is specific need to write signature to disk(cache) we should abandon this feature.

Opened an issue https://github.com/notaryproject/notation/issues/379